pyproject.toml

```diff
@@ -9,9 +9,10 @@ authors = [
 readme = "README.md"
 
 [tool.poetry.dependencies]
-python = "^3.11"
+python = "^3.12"
 pyyaml = "^6.0.1"
 xarray = "^2024.3.0"
+pyarrow = "^16.1.0"
```

Comment:
I don't remember why I added pyarrow, but if it isn't needed we should leave it out. I don't remember needing to interface a subtle C++ library anywhere.  I've taken it out because my tests run without it.